--- a/setup.py
+++ b/setup.py
@@ -151,9 +151,9 @@
 
 platform = sys.platform
 
-# Detect Python for android project (http://github.com/kivy/python-for-android)
-ndkplatform = environ.get('NDKPLATFORM')
-if ndkplatform is not None and environ.get('LIBLINK'):
+# Detect BuildEnvironment of DroidBuilder (http://github.com/kmedya-dev/droidbuilder)
+ndkplatform = environ.get('ANDROID_NDK_ROOT')
+if ndkplatform is not None:
     platform = 'android'
 kivy_ios_root = environ.get('KIVYIOSROOT', None)
 if kivy_ios_root is not None:
@@ -248,7 +248,7 @@
 # This determines whether Cython specific functionality may be used.
 can_use_cython = True
 
-if platform in ('ios', 'android'):
+if platform in ('ios'):
     # NEVER use or declare cython on these platforms
     print('Not using cython on %s' % platform)
     can_use_cython = False
@@ -627,6 +627,10 @@
         flags['include_dirs'] += [sysroot]
         flags['extra_compile_args'] += ['-isysroot', sysroot]
         flags['extra_link_args'] += ['-isysroot', sysroot]
+    elif c_options['use_android']:
+        sysroot = environ.get('SYSROOT')
+        if not sysroot:
+            raise Exception('SYSROOT environment variable is not set for Android build')
     elif platform.startswith('freebsd'):
         flags['include_dirs'] += [join(
             environ.get('LOCALBASE', '/usr/local'), 'include')]
@@ -683,8 +687,8 @@
         flags['library_dirs'] = ['/usr/X11R6/lib']
         flags['libraries'] = ['GL']
     elif platform == 'android':
-        flags['include_dirs'] = [join(ndkplatform, 'usr', 'include')]
-        flags['library_dirs'] = [join(ndkplatform, 'usr', 'lib')]
+        flags['include_dirs'] = [join(environ.get('SYSROOT'), 'usr', 'include')]
+        flags['library_dirs'] = [join(environ.get('SYSROOT'), 'usr', 'lib')]
         flags['libraries'] = ['GLESv2']
     elif platform == 'rpi':
 
